python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_auto_materialize_policy.py

```diff
@@ -74,6 +76,7 @@ def test_with_rules():
             AutoMaterializeRule.skip_on_parent_outdated(),
             AutoMaterializeRule.skip_on_parent_missing(),
             AutoMaterializeRule.materialize_on_required_for_freshness(),
+            AutoMaterializeRule.discard_on_max_materializations_exceeded(limit=1),
```

Comment:
I'm somewhat wary of the impact here, as this means users building policies from scratch will not have this safeguard built-in.

Depending on how we envision the ideal API for customizing these rules, that could be fine, or not. In light of the discussion this morning about shying away from adding more discard rules, I think we can probably put this PR on hold until the point that we evaluate how likely it is that a user will be compelled to construct a policy like:

```python
AutoMaterializePolicy(
        rules={AutoMaterializeRule.materialize_on_parent_updated()}
    )
```

instead of 

```python
AutoMaterializePolicy.eager().without_rules(...)
```
(or something else)

apologies for the thrash here, as overall this change seems positive, but just harder to reverse than I'd like

Reply:
This makes sense, I'm happy to punt on this PR